packages/sanity/src/tasks/src/tasks/components/list/TasksListItem.tsx

```diff
@@ -0,0 +1,143 @@
+  unstable_useValuePreview as useValuePreview,
+  useDateTimeFormat,
+  useUser,
+  useSchema,
+  useClient,
+} from '../../../../../core'
+import {SanityDocument} from '@sanity/types'
+
+interface TasksListItemProps {
+  title?: string
+  assignedTo?: string
+  dueBy?: string
+  onSelect: () => void
+  target?: any
+}
+
+export const ThreadCard = styled(Card).attrs<CardProps>(({tone}) => ({
+  padding: 3,
+  radius: 3,
+  sizing: 'border',
+  tone: tone || 'transparent',
+}))<CardProps>`
+  // ...
+`
+
+const Title = styled(Text)`
+  :hover {
+    text-decoration: underline;
+  }
+`
+
+const SKELETON_INLINE_STYLE: React.CSSProperties = {width: '50%'}
+
+function AssignedToSection({userId}: {userId: string}) {
+  const [user] = useUser(userId)
+
+  const name = user?.displayName ? (
+    <Text muted size={1} weight="medium" textOverflow="ellipsis" title={user.displayName}>
+      {user.displayName}
+    </Text>
+  ) : (
+    <TextSkeleton size={1} style={SKELETON_INLINE_STYLE} />
+  )
+
+  return (
+    <Flex align="center" gap={1}>
+      <UserIcon />
+      {name}
+    </Flex>
+  )
+}
+
+function getTargetDocumentMeta(target: any) {
+  if (!target?.document._ref) {
+    return undefined
+  }
+
+  return {
+    _ref: target?.document._ref,
+    _type: target?.documentType,
+  }
+}
+
+export function TasksListItem({assignedTo, title, dueBy, target, onSelect}: TasksListItemProps) {
+  const dateFormatter = useDateTimeFormat({
+    dateStyle: 'medium',
+  })
+  const dueByeDisplayValue = useMemo<string | undefined>(() => {
+    return dueBy ? dateFormatter.format(new Date(dueBy)) : undefined
+  }, [dateFormatter, dueBy])
+
+  const targetDocument = useMemo(() => getTargetDocumentMeta(target), [target])
+
+  return (
+    <ThreadCard tone={undefined}>
+      <Stack space={2}>
+        <Title size={1} weight="semibold" onClick={onSelect}>
+          {title || 'Untitled'}
+        </Title>
+        {assignedTo && <AssignedToSection userId={assignedTo} />}
+        {dueByeDisplayValue && (
+          <Flex align="center" gap={1}>
+            <CalendarIcon />
+            <Text size={1} muted>
+              {dueByeDisplayValue}
+            </Text>
+          </Flex>
+        )}
+        {targetDocument && (
+          <DocumentPreview documentId={targetDocument._ref} documentType={targetDocument._type} />
+        )}
+      </Stack>
+    </ThreadCard>
+  )
+}
+
+function DocumentPreview({documentId, documentType}: {documentId: string; documentType: string}) {
+  const [documentValue, setDocumentValue] = useState<SanityDocument | null>(null)
+  const client = useClient({
+    apiVersion: '2023-01-01',
+  }).withConfig({
+    perspective: 'previewDrafts',
+  })
+  const schema = useSchema()
+  const documentSchema = schema.get(documentType)
+  const {isLoading, value} = useValuePreview({
+    enabled: true,
+    schemaType: documentSchema,
+    value: {
+      _id: documentValue?._originalId ?? documentId,
+    },
+  })
+
+  useEffect(() => {
```

Comment:
Hey @sjelfull why do we need to fetch the document here? Is there a difference with the documentValue?._originalId compared to documentId?